cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -71,6 +81,160 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  /**
+   * Get {@link LocalDate} from field of logical type date.
+   *
+   * @param fieldName date field to get.
+   * @return value of the field as an object of {@link LocalDate}, returns null if given field has null value
+   * @throws IllegalArgumentException if the field name is not present in schema or the field provided
+   *                                  is not of {@link Schema.LogicalType#DATE} type.
+   */
+  public LocalDate getDate(String fieldName) throws IllegalArgumentException {
+    validateAndGetLogicalType(fieldName, schema.getField(fieldName),
+                              new HashSet<>(Collections.singletonList(Schema.LogicalType.DATE.name())));
```

Comment:
No need to `new HashSet`. Use `EnumSet` instead (see comment below about using the enum instead of String).